src/tools/rbd/action/Bench.cc

```diff
@@ -162,7 +167,13 @@ struct rbd_bencher {
     } else {
       c = new librbd::RBD::AioCompletion((void *)(new bencher_completer(this, NULL)),
 					 rbd_bencher_completion);
-      image->aio_write2(off, len, write_bl, c, op_flags);
+      if (io_type == IO_TYPE_WRITE_ZEROES) {
+        image->aio_write_zeroes(off, len, c, RBD_WRITE_ZEROES_FLAG_THICK_PROVISION, op_flags);
```

Comment:
I agree that the write_zeros is not very useful, especially with the other PR which lets you choose what pattern of data is being written (for testing Erasure Coding it can be quite useful to be able to write 0's to parts of stripe). I'll remove it from this PR and just add support for discard. I'll leave providing a more programmatic way of issuing I/Os and exercising the flags better to a future PR.